chore: remove js dep and added wasm build ci

abdelaziz-mahdy opened the pull request:
fix https://github.com/media-kit/media-kit/issues/853 and added ci to ensure its working 

now it doesnt work since full screen and clear are not found, nor can i figure out the blob new syntax

sgehrman commented:
I'm waiting on this.  Please fix.

abdelaziz-mahdy commented:
I don't know the migration for the mentioned problems I left the pr for other people to fix the rest of the problems and if I got time to fix it myself, but no promises.

abdelaziz-mahdy commented:
i cant figure how to migrate `media_kit/lib/src/player/web` sadly

hasanm08 commented:
> I don't know the migration for the mentioned problems I left the pr for other people to fix the rest of the problems and if I got time to fix it myself, but no promises.
> 
> will wait for [dart-lang/web#259 (comment)](https://github.com/dart-lang/web/issues/259#issuecomment-2189509263) for fullscreen part, for the rest of the stuff research is still needed

that item fixed on web 1.0.0

abdelaziz-mahdy commented:
> > I don't know the migration for the mentioned problems I left the pr for other people to fix the rest of the problems and if I got time to fix it myself, but no promises.
> > will wait for [dart-lang/web#259 (comment)](https://github.com/dart-lang/web/issues/259#issuecomment-2189509263) for fullscreen part, for the rest of the stuff research is still needed
> 
> that item fixed on web 1.0.0

i already did the full screen part and blob in the last commits

abdelaziz-mahdy commented:
> hello? what is taking so long?

The migration for the web player is not easy @alexmercerind if you can help with that that will be awesome

Matzu89 commented:
Honestly, you would be one of the few with wasm support. A lot of packages have no support for this yet, so it seems.

abdelaziz-mahdy commented:
> Honestly, you would be one of the few with wasm support. A lot of packages have no support for this yet, so it seems. 

The migration is not the same for every package, so yeah it will take a while to migrate the web player correctly and accurately

abdelaziz-mahdy commented:
@TheComputerM i reverted the changes for the player and fixed merge conflicts, this pr now only includes changes for the video player controller and the other files leaving the backend for the player for you

TheComputerM commented:
> @TheComputerM i reverted the changes for the player and fixed merge conflicts, this pr now only includes changes for the video player controller and the other files leaving the backend for the player for you

Alright, I'll make the changes required after this PR gets merged.

alexmercerind commented on media_kit/lib/src/player/web/utils/hls.dart:
```diff
@@ -4,6 +4,7 @@
 /// All rights reserved.
 /// Use of this source code is governed by MIT license that can be found in the LICENSE file.
 import 'dart:async';
+// import 'package:web/web.dart' as html;
```
nit: remove redundant comment

alexmercerind commented on media_kit_video/lib/src/video/video_web.dart:
```diff
@@ -450,7 +452,7 @@ class VideoState extends State<Video> with WidgetsBindingObserver {
 Future<void> defaultEnterNativeFullscreen() async {
   try {
-    await document.documentElement?.requestFullscreen();
+    await document.documentElement?.requestFullscreen().toDart;
```
Does this work?

abdelaziz-mahdy replied:
just tested and yes it works on javascript code (cant compile to wasm since the player is not migrated yet but if it works for javascript it should work for wasm)

alexmercerind commented on media_kit_video/lib/src/video/video_web.dart:
```diff
@@ -460,7 +462,7 @@ Future<void> defaultEnterNativeFullscreen() async {
 Future<void> defaultExitNativeFullscreen() async {
   try {
-    document.exitFullscreen();
+    await document.exitFullscreen().toDart;
```
Does this work?

abdelaziz-mahdy replied:
just tested and yes it works on javascript code (cant compile to wasm since the player is not migrated yet but if it works for javascript it should work for wasm)

alexmercerind reviewed (approved):
LGTM otherwise!

abdelaziz-mahdy commented:
I tested the changes on 
https://pub.dev/packages/universal_video_controls/
And it works correctly there I can test again to be sure, but it should work anyway

alexmercerind commented:
> I tested the changes on https://pub.dev/packages/universal_video_controls/ And it works correctly there I can test again to be sure, but it should work anyway

Can you credit media_kit more appropriately in the package e.g. hyperlinks etc.

abdelaziz-mahdy commented:
> > I tested the changes on https://pub.dev/packages/universal_video_controls/ And it works correctly there I can test again to be sure, but it should work anyway
> 
> Can you credit media_kit more appropriately in the package e.g. hyperlinks etc. ?

Will do, sorry

abdelaziz-mahdy commented:
> > I tested the changes on https://pub.dev/packages/universal_video_controls/ And it works correctly there I can test again to be sure, but it should work anyway
> 
> Can you credit media_kit more appropriately in the package e.g. hyperlinks etc. ?

i made it a hyper link if you think of any other changes let me know

abdelaziz-mahdy commented:
@alexmercerind should we leave the wasm ci? since it will always fail?

EricApostal commented:
I was able to complete the migration to a compilable state based on this fork. How should I try and get my changes under review? I can fork the main repo and implement my changes directory, or just PR to https://github.com/abdelaziz-mahdy/media_kit/. I haven't ran the test CI but it seems to be running okay. It includes player/web, as well as HLS, etc.

@alexmercerind 

![image](https://github.com/user-attachments/assets/5559c1b6-29b5-44c1-8300-2949b89e7d52)

abdelaziz-mahdy commented:
> I was able to complete the migration to a compilable state based on this fork. How should I try and get my changes under review? I can fork the main repo and implement my changes directly, or just PR to https://github.com/abdelaziz-mahdy/media_kit/. I haven't ran the test CI but it seems to be running okay. It includes player/web, as well as HLS, etc.
> 
> @alexmercerind 
> 
> ![image](https://github.com/user-attachments/assets/5559c1b6-29b5-44c1-8300-2949b89e7d52)
> 

I think you can open a pr into my branch and I don't mind reviewing it and merging it to this pr, which will make full migration to wasm and will run the ci

EricApostal commented:
I apologize- there might be a delay on that. I tried the test and for some reason, while the video player example appears to work, the `media_kit_test` project is having issues when initializing the controller. Don't wait on me for this, worse comes to worst I just PR after this is through.

abdelaziz-mahdy commented:
> I apologize- there might be a delay on that. I tried the test and for some reason, while the video player example appears to work, the `media_kit_test` project is having issues when initializing the controller. Don't wait on me for this, worse comes to worst I just PR after this is through.

ok will merge this pr then and you can open a new pr when you finish the migration
